test/storage.test.js

```diff
@@ -129,6 +132,169 @@ describe('storage.js', () => {
+                    return 'https://example.com/preSignedUrl';
+                },
+                download : () => {
+                    console.log('Fake file downloaded');
+                }
+            });
+            const {getAsset} = mockRequire.reRequire("../lib/storage");
+            mockFs({'./in/fakeSource/filePath': {}});
+            // This is mocked to pass the fileExistsCheck
+            fs.writeFileSync('./in/fakeSource/filePath/source','something');
+            assert.ok(fs.existsSync(directory));
+            const source = await getAsset(assetReference, directory, name, disableSourceDownload);
+            console.log(source);
+            assert.strictEqual(source.name, 'source');
+            assert.strictEqual(source.path, 'in/fakeSource/filePath/source');
+            assert.strictEqual(source.params.url, 'https://example.com/preSignedUrl');
+        });
+
+        it('should not generate presignurl when source url is http  when disabled download is true', async () => {
+            const assetReference = {
+                url: 'http://nondatauriurl.com/example.png'
+            };
+            
+            mockFs.restore();
+            mockRequire('../lib/storage/datauri', {
+                getPreSignedUrl : (source) => {
+                    console.log('Mocked source : ' + source);
+                    return 'https://example.com/preSignedUrl';
+                },
+                download : () => {
+                    console.log('Fake file downloaded');
+                }
+            });
+            
+            const directory = './in/fakeSource/filePath';
+            const name = 'source';
+            const disableSourceDownload = true;
+
+            const {getAsset} = mockRequire.reRequire("../lib/storage");
+            const source = await getAsset(assetReference, directory, name, disableSourceDownload);
+            assert.strictEqual(source.name, 'source');
+            assert.strictEqual(source.path, 'in/fakeSource/filePath/source');
+            assert.strictEqual(source.params.url, 'http://nondatauriurl.com/example.png');
+        });
+
+        it("should fail to generate preSignUrl for  empty data uri", async() =>{
+            const assetReference = {
+                url: "data:,"
+            };
+            const directory = './in/fakeSource/filePath';
+            const name = 'source';
+            const disableSourceDownload = true;
+            mockFs({ './in/fakeSource/filePath': {} });
+            assert.ok(fs.existsSync(directory));
+            try {
+                await getAsset(assetReference, directory, name, disableSourceDownload);
+            } catch (e) {
+                assert.strictEqual(e.name, 'SourceUnsupportedError');
+                assert.strictEqual(e.message, 'Invalid or missing local file in/fakeSource/filePath/source');
+            }
+        });
+
+        it("should fail to generate preSignUrl for non existent file", async() =>{
```

Comment:
what is the difference between this test and the one above?

Reply:
The difference is first one takes care of empty "data:," meaning file is created but has no content, second one something went wrong on download or file does not exist